applications/vanilla/views/post/comment.php

```diff
@@ -45,7 +45,7 @@
 
                     echo '<span class="'.$CancelClass.'">';
                     echo anchor($CancelText, '/');
-                    if ($CategoryID = $this->data('Discussion.CategoryID')) {
+                    if ($CategoryID = $this->data('Discussion.CategoryID') && c('Vanilla.Categories.Use', true)) {
```

Comment:
I've changed that if condition some more, because I think using `if  ($a = $b)` is confusing and should be avoided. It is a mixture of `$a = $b; if ($b)` but I would always expect to see a "==" or "===" in an if condition. I don't think that is against the coding style, but when looking at it, my first reaction have been that this must be a typo.